js/kanji.js

```diff
@@ -150,6 +151,52 @@ window.Kanji =  {
     }
   },
 
+  _initialSelectAllHide: function() {
+    $(".category-select-all").each(function() {
+      $(this).hide();
+    });
+    $(".category-deselect-all").each(function() {
+      $(this).hide();
+    });
+    $(".category-select-all-show").each(function() {
+      $(this).show();
+    });
+  },
+
+  __selectClicker: function() {
+    if (!$(this).hasClass('selected')) {
+      $(this).trigger("click");
+    }
+  },
+
+  __deselectClicker: function() {
+    if ($(this).hasClass('selected')) {
+      $(this).trigger("click");
+    }
+  },
+
+  _handleSelectAll: function() {
+    $(".category-select-all,.category-deselect-all").off("click"); // remove previous event handlers
+
+    $(".category-select-all").on('click', function() {
```

Comment:
This and the click event below should be two separate handling functions. One that handles select, one for deselect. 

It seems like the logic is really similar be tween the two. Maybe there is a way to make it one functions and pass in whether or not you are selecting/deselecting to change behavior?